File: Samples/BasicApplication/BasicApplication/BasicApplication.Script/Imports/ServiceContracts/Northwind.CategoryRow.cs

namespace BasicApplication.Northwind
{
    using Serenity;
    using Serenity.ComponentModel;
    using System;
    using System.Collections;
    using System.Collections.Generic;
    using System.ComponentModel;
    using System.Runtime.CompilerServices;

    [Imported, Serializable, PreserveMemberCase]
    public partial class CategoryRow
    {
        public Int32? CategoryID { get; set; }
        public String CategoryName { get; set; }
        public String Description { get; set; }
        public byte[] Picture { get; set; }
    
        [Imported, PreserveMemberCase]
        public static class Fields
        {
            [InlineConstant] public const string CategoryID = "CategoryID";
            [InlineConstant] public const string CategoryName = "CategoryName";
            [InlineConstant] public const string Description = "Description";
            [InlineConstant] public const string Picture = "Picture";
        }
    }
    
}

File: common-features/src/demo.northwind/Modules/ServerTypes/Demo/Northwind.CategoryRow.ts
import { getLookup, getLookupAsync, fieldsProxy } from "@serenity-is/corelib";

export interface CategoryRow {
    CategoryID?: number;
    CategoryName?: string;
    Description?: string;
    Picture?: number[];
}

export abstract class CategoryRow {
    static readonly idProperty = 'CategoryID';
    static readonly nameProperty = 'CategoryName';
    static readonly localTextPrefix = 'Northwind.Category';
    static readonly lookupKey = 'Northwind.Category';

    /** @deprecated use getLookupAsync instead */
    static getLookup() { return getLookup<CategoryRow>('Northwind.Category') }
    static async getLookupAsync() { return getLookupAsync<CategoryRow>('Northwind.Category') }

    static readonly deletePermission = 'Northwind:General';
    static readonly insertPermission = 'Northwind:General';
    static readonly readPermission = 'Northwind:General';
    static readonly updatePermission = 'Northwind:General';

    static readonly Fields = fieldsProxy<CategoryRow>();
}